libcob/fileio.c

```diff
@@ -5429,6 +5464,7 @@ indexed_read_next (cob_file *f, const int read_opts)
 		} else {
 			p->key.size = (cob_dbtsize_t) bdb_keylen(f,p->key_index);
 			p->key.data = p->last_readkey[p->key_index];
+			bdb_setkeycol (f, p->key_index);
```

Comment:
when placing this into each `cob_key` you can direclty do `p->key.app_data = f->keys[p->key_index]->collation` here